apps/chat/src/utils/app/data/application-service.ts

```diff
@@ -0,0 +1,123 @@
+import { Observable, map } from 'rxjs';
+
+import {
+  ApplicationDetailsResponse,
+  ApplicationListItemModel,
+  ApplicationListResponseModel,
+  ApplicationMoveModel,
+  CreateApplicationModel,
+} from '@/src/types/applications';
+import { EntityType } from '@/src/types/common';
+import { HTTPMethod } from '@/src/types/http';
+import { DialAIEntityModel } from '@/src/types/models';
+
+import { ApiUtils } from '../../server/api';
+import { constructPath } from '../file';
+import { BucketService } from './bucket-service';
+
+export class ApplicationService {
+  public static create(
+    applicationName: string,
+    applicationData: CreateApplicationModel,
+  ): Observable<ApplicationListItemModel> {
+    const bucket = BucketService.getBucket();
+    return ApiUtils.request(
+      constructPath(
+        'api',
+        'applications',
+        bucket,
+        ApiUtils.encodeApiUrl(applicationName),
+      ),
+      {
+        method: 'POST',
+        body: JSON.stringify(applicationData),
+      },
+    );
+  }
+
+  public static edit(
+    applicationData: CreateApplicationModel,
+  ): Observable<ApplicationListItemModel> {
+    const bucket = BucketService.getBucket();
+    return ApiUtils.request(
+      constructPath(
+        'api',
+        'applications',
+        bucket,
+        ApiUtils.encodeApiUrl(applicationData.display_name),
+      ),
+      {
+        method: 'PUT',
+        body: JSON.stringify(applicationData),
+      },
+    );
+  }
+
+  public static move(
+    data: ApplicationMoveModel,
+  ): Observable<ApplicationMoveModel> {
+    const bucket = BucketService.getBucket();
+    return ApiUtils.request('api/ops/resource/move', {
+      method: 'POST',
+      body: JSON.stringify({
+        sourceUrl: constructPath(
+          'applications',
+          bucket,
+          ApiUtils.encodeApiUrl(data.sourceUrl),
+        ),
+        destinationUrl: constructPath(
+          'applications',
+          bucket,
+          ApiUtils.encodeApiUrl(data.destinationUrl),
+        ),
+        overwrite: data.overwrite,
+      }),
+    });
+  }
+
+  public static delete(applicationUrl: string): Observable<string> {
+    const bucket = BucketService.getBucket();
+    return ApiUtils.request(
+      constructPath(
+        'api',
+        'applications',
+        bucket,
+        ApiUtils.encodeApiUrl(applicationUrl),
+      ),
+      {
+        method: 'DELETE',
+      },
+    );
+  }
+
+  public static listing(): Observable<ApplicationListResponseModel[]> {
+    const bucket = BucketService.getBucket();
+    return ApiUtils.request(
+      constructPath('api', 'application', 'listing', bucket),
+      {
+        method: 'GET',
```

Comment:
```suggestion
        method: HTTPMethod.GET,
```
pay attention on [this comment](https://github.com/epam/ai-dial-chat/pull/1851/files#r1711340231)